src/encoder.ts

```diff
@@ -20,18 +20,20 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
+import * as _ from 'lodash';
+
 export function dateToTimestampProto(timeString) {
   if (typeof timeString === 'undefined') {
     return;
   }
   let date = new Date(timeString);
   let seconds = Math.floor(date.getTime() / 1000);
-  let nanoString = timeString.substring(20, timeString.length - 1);
-
-  if (nanoString.length === 3) {
-    nanoString = `${nanoString}000000`;
-  } else if (nanoString.length === 6)  {
-    nanoString = `${nanoString}000`;
+  let nanoString;
+  if (timeString.length <= 20) {
+    nanoString = '000000000';
+  } else {
+    nanoString = timeString.substring(20, timeString.length - 1);
+    nanoString += _.reduce(new Array(9 - nanoString.length), i => {return i + '0';}, '');
```

Comment:
Woah... String -> array -> reduce -> parseInt? It may be the c programmer in me, but string manipulation of numbers feels really wonky.

why not something like

```typescript
let nanos: 0;
if (timeString.length > 20) {
  const nanoString = timeString.substring(20, timeString.length - 1);
  const trailingZeros = 9 - nanoString.length;
  nanos = parseInt(nanoString, 10) * Math.pow(10, trailingZeroes);
}
```